go/vt/vtctl/workflow/traffic_switcher.go

```diff
@@ -548,7 +548,9 @@ func (ts *trafficSwitcher) removeSourceTables(ctx context.Context, removalType T
 				DisableForeignKeyChecks: true,
 			})
 			if err != nil {
-				if mysqlErr, ok := err.(*sqlerror.SQLError); ok && mysqlErr.Num == sqlerror.ERNoSuchTable {
+				// The error is a gRPC status.Error, so we need to convert it to an sqlerror.SQLError.
+				if mysqlErr, ok := sqlerror.NewSQLErrorFromError(err).(*sqlerror.SQLError); ok &&
+					(mysqlErr.Num == sqlerror.ERNoSuchTable || mysqlErr.Num == sqlerror.ERBadTable) {
 					ts.Logger().Warningf("%s: Table %s did not exist when attempting to remove it", topoproto.TabletAliasString(source.GetPrimary().GetAlias()), tableName)
 					return nil
```

Comment:
We should not return here, right? Otherwise we will not delete subsequent participating tables. Same below.

Reply:
Doh! Good catch!

Reply:
Fixed in: https://github.com/vitessio/vitess/pull/16505/commits/f83427050ab9747f19f87502d4bff451c00f9cab